### PR TITLE
RND-5227: Site section tabs

### PR DESCRIPTION
**Pull request by BrettJephson**

### CODE REVIEW

File: packages/gitbook/src/app/(site)/fetch.ts [hunk 1]
```diff
@@ -129,20 +139,44 @@ async function resolvePage(
...
 /**
  * Fetch the structure of an organization site.
- * This includes the site and its spaces.
+ * This includes the site and its sections and spaces.
  */
 async function fetchSiteStructure(args: {
     organizationId: string;
     siteId: string;
     siteShareKey: string | undefined;
+    siteSpaceId: string | undefined;
```
Comment (taranvohra) [inline packages/gitbook/src/app/(site)/fetch.ts:148]:
```suggestion

```
We don't need this

File: packages/gitbook/src/app/(site)/fetch.ts [hunk 2]
```diff
@@ -68,6 +77,7 @@ export async function fetchPageData(params: PagePathParams | PageIdParams) {
             organizationId: content.organizationId,
             siteId: content.siteId,
             siteShareKey: content.siteShareKey,
+            siteSpaceId: content.siteSpaceId,
```
Comment (taranvohra) [inline packages/gitbook/src/app/(site)/fetch.ts:80]: Same thing, we shouldn't be passing `siteSpaceId`

File: packages/gitbook/src/app/(site)/fetch.ts [hunk 3]
```diff
@@ -33,19 +33,28 @@ export async function fetchContentData() {
...
                 siteShareKey: content.siteShareKey,
+                siteSpaceId: content.siteSpaceId,
             }),
         ]);
 
     const site = siteStructure.site;
-    const spaces = siteStructure.spaces;
+
+    const sections = siteStructure.sections;
+    const section = siteStructure.sections?.find((section) => section.id === content.siteSectionId);
```
Comment (taranvohra) [inline packages/gitbook/src/app/(site)/fetch.ts:43]: If it's a sections site then `section` should ALWAYS be defined else we try. I think you can put a condition that 
```js
if(sections) {
   assertIsDefined(section);
}
```

File: packages/gitbook/src/app/(site)/fetch.ts [hunk 4]
```diff
@@ -33,19 +33,28 @@ export async function fetchContentData() {
...
-    const spaceRelativeToParent = spaces.find((space) => space.id === content.spaceId);
+    const spaceRelativeToParent = spaces?.find((space) => space.id === content.spaceId);
 
     return {
         content,
         contentTarget,
         space: spaceRelativeToParent ?? space,
         pages,
+        sections,
+        section,
```
Comment (taranvohra) [inline packages/gitbook/src/app/(site)/fetch.ts:57]: On that same note the return typing should also indicate that `section` would always exist if site has sections. We could maybe refactor this to be

```js
return {
  ...,
 sections: sections ? { list: sections, section } : undefined
}
```

File: packages/gitbook/src/components/SiteSectionTabs/SiteSectionTab.tsx
Comment (taranvohra) [inline packages/gitbook/src/components/SiteSectionTabs/SiteSectionTab.tsx:1]: Lets also add few comments to these components

**Review by taranvohra — APPROVED** [review]
Looks good overall, minor suggestions to improve the typing DX

File: packages/gitbook/src/app/(site)/fetch.ts (hunk 1 above)
Comment (BrettJephson) [reply packages/gitbook/src/app/(site)/fetch.ts:148]: Forgot to remove it. Thanks!

File: packages/gitbook/src/app/(site)/fetch.ts [hunk 5]
```diff
@@ -54,6 +64,19 @@ export async function fetchContentData() {
     };
 }
 
+function getSiteSectionsList(siteSectionId?: string, sections?: SiteSection[] | null) {
+    if (!sections) {
+        return null;
+    }
+    const section = sections?.find((section) => section.id === siteSectionId);
```
Comment (taranvohra) [inline packages/gitbook/src/app/(site)/fetch.ts:71]:
```suggestion
    const section = sections.find((section) => section.id === siteSectionId);
```

File: packages/gitbook/src/app/(site)/fetch.ts [hunk 6]
```diff
@@ -54,6 +64,19 @@ export async function fetchContentData() {
     };
 }
 
+function getSiteSectionsList(siteSectionId?: string, sections?: SiteSection[] | null) {
```
Comment (taranvohra) [inline packages/gitbook/src/app/(site)/fetch.ts:67]:
```suggestion
function parseSiteSectionsList(siteSectionId?: string, sections?: SiteSection[] | null) {
```

Comment (taranvohra) [inline packages/gitbook/src/app/(site)/fetch.ts:67]: This signature is a bit confusing. Why not `(sections: SiteSection[], siteSectionId: string)` and then do `const siteSections = content.siteSectionId && siteStructure.sections ? parseSiteSections(...) : null`

**Review by taranvohra — APPROVED** [review]

File: packages/gitbook/src/app/(site)/fetch.ts (hunk 6 above)
Comment (BrettJephson) [reply packages/gitbook/src/app/(site)/fetch.ts:67]: Good point. I was thinking I'd encapsulate it in the function but it definitely makes the function signature confusing.

**Pull request merged**